src/annotator/anchoring/pdf.js

```diff
@@ -42,21 +42,31 @@ export const RenderingStates = {
 /**
  * Map of page index to page text content.
  *
- * @type {Record<number,Promise<string> | undefined>}
+ * @type {Map<number,Promise<string> | undefined>}
  */
-let pageTextCache = {};
+const pageTextCache = new Map();
 
 /**
- * A cache that maps a `(quote, text offset in document)` key to a specific
+ * A cache that maps a `{quote}:{offset}` key to a specific
```

Comment:
This is a good idea to reduce the number of _collisions_. Several annotation could have the same quote but could be anchored in different places.

However, I feel we should reduce further the possibility of hitting the wrong cache by adding the start and end offset of the anchored text: 

```
function quotePositionCacheKey(quote, start, end) {
  return `${quote}:${start}:${end}`;
```

Imaging two annotations that start in the same position, but finish in different one, with the proposed change, each will have their own cache.

Furthermore, we have different offsets systems depending of the PDF.js version. Adding as much information as possible in the cache key, could further reduce possible _collisions_.

Reply:
We discussed on Slack, but to summarize: The end offset is determined by the start offset and quote length, so in this case it would always be the same for given `quote` and `start` arguments.